src/cls.ts

```diff
@@ -17,9 +17,28 @@
 import * as CLS from 'continuation-local-storage';
 import * as semver from 'semver';
 
-import {SpanData} from './span-data';
+import {SpanDataType} from './constants';
+import {UNCORRELATED_SPAN, UNTRACED_SPAN} from './span-data';
+import {Trace, TraceSpan} from './trace';
+
+/**
+ * This type represents the minimal information to store in continuation-local
+ * storage for a request. We store either a root span corresponding to the
+ * request, or a sentinel value (UNCORRELATED_SPAN or UNTRACED_SPAN) that tells
+ * us that the request is not being traced (with the exact sentinel value
+ * specifying whether this is on purpose or by accident, respectively).
+ *
+ * When we store an actual root span, the only information we need is its
+ * current trace/span fields.
+ */
+export type RootContext = ({
+  readonly span: TraceSpan;
+  readonly trace: Trace;
+  readonly type: SpanDataType.ROOT;
+}|{
```

Comment:
nit: it would be good to name these anonymous types. Something like `RealRootContext` and `PhantomRootContext`?

Reply:
Done